src/android/UnityAdsPlugin.java

```diff
@@ -103,60 +168,19 @@ private void initialize(JSONArray args, CallbackContext callbackContext) {
             }
 
             UnityAds.setDebugMode(debugMode);
-            UnityAds.initialize(cordova.getActivity(), gameId, this, testMode);
+            UnityAds.initialize(getApplicationContext(), gameId, testMode, this);
         }
 
         @Override
-        public void onUnityAdsReady(String s) {
-            if (!isInitialized) {
-                isInitialized = true;
-                callbackID.success();
-            }
-        }
-
-        @Override
-        public void onUnityAdsStart(String s) {
-
-        }
-
-        @Override
-        public void onUnityAdsFinish(String s, UnityAds.FinishState finishState) {
-            if (finishState == UnityAds.FinishState.COMPLETED) {
-                callbackID.success();
-            } else if (finishState == UnityAds.FinishState.SKIPPED) {
-                callbackID.error("VIDEO_SKIPPED");
-            } else {
-                callbackID.error("DID FINISH WITH ERROR");
-            }
-        }
-
-        @Override
-        public void onUnityAdsError(UnityAds.UnityAdsError error, String message) {
-            String msg;
-
-            Log.d(TAG, String.format("%s", "onUnityAdsError"));
-            Log.d(TAG, String.format("%s", message));
-
-            if(error == UnityAds.UnityAdsError.NOT_INITIALIZED){
-                msg = String.format("[\"%s\",\"%s\"]", message, "NOT_INITIALIZED");
-            }else if(error == UnityAds.UnityAdsError.INITIALIZE_FAILED){
-                msg = String.format("[\"%s\",\"%s\"]", message, "INITIALIZE_FAILED");
-            }else{
-                msg = String.format("[\"%s\",\"%s\"]", message, "INTERNAL_ERROR");
-            }
-
-            callbackID.error(msg);
-        }
-
-        @Override
-        public void onUnityAdsClick(String s) {
+        public void onInitializationComplete() {
             callbackID.success();
+            this.onInitialized(this.args);
         }
 
         @Override
-        public void onUnityAdsPlacementStateChanged(String s, UnityAds.PlacementState placementState, UnityAds.PlacementState placementState1) {
-
+        public void onInitializationFailed(UnityAds.UnityAdsInitializationError error, String message) {
+            Log.w(TAG, "onInitializationFailed" + message);
+            callbackId.error(String.format("[\"%s\",\"%s\"]", message, error.name()));
```

Comment:
The app is not compiling because of this typo

```suggestion
            callbackID.error(String.format("[\"%s\",\"%s\"]", message, error.name()));
```